PR-10 review comments

sarath-sasi-panickaveetil opened the pull request:
> Static path removal
> unfiltered icon added
> Hardcoded JSON added

sarath-sasi-panickaveetil committed:
Order Details component
> Icon static paths removed
> Customized Data table and Card components

sarath-sasi-panickaveetil committed:
PR-10 Review Suggestions
> Loader icon alignement
> hardcoded JSON added to display data against orderStatus and OrderOMSSatus

raj-kumar-litmus commented on src/components/Common/Card.tsx:
```diff
@@ -1,26 +1,80 @@
-          </div>
-        ) : (
-          <div key={ind} className="w-6/12 pl-4 flex">
+      <div className="w-3/12 pl-4 pr-4 text-center items-center border-r flex flex-col justify-center">
```
Since u modified an existing component, can u confirm if the existing implementation of this component (by Aditya) is still working as expected ?

sarath-sasi-panickaveetil replied:
Confirmed that it does not break existing components.

raj-kumar-litmus commented on src/components/charts/BarChart.tsx:
```diff
@@ -114,15 +120,16 @@ const BarChart = () => {
-    //   starttime: startDateTime,
-    //   channel: channel,
-    // };
+    const params = {
```
I commented this out because I got typescript error for unused variable.
how come you didn't get the same error ?

sarath-sasi-panickaveetil replied:
ran "npm run build" which does not give any error. Can you accept this for now since 'params' must be used once API implementation takes place?

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+// import {
```
Remove commented lines if not needed

sarath-sasi-panickaveetil replied:
Commented these lines since we are using hardcoded JSON for now. Will uncomment them once we use actual APIs.

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+
+const OrderDetails: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [orderData, setOrderData] = useState<OrderData>({});
```
You would have got a typescript warning in this line.

sarath-sasi-panickaveetil replied:
added Record<string, never> for empty object

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+const OrderDetails: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [orderData, setOrderData] = useState<OrderData>({});
+  const [omsOrderStatus, setOmsOrderStatus] = useState<OmsOrderStatus>({});
```
Same comment as above.

sarath-sasi-panickaveetil replied:
added Record<string, never> for empty object

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+    //   `${URL_ORDER_DETAILS}/${orderId}`,
+    //   {},
+    // );
+    console.log(orderId);
```
remove console.log before raising MR.

sarath-sasi-panickaveetil replied:
removed

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+                className="ml-2 cursor-pointer"
+                alt="order-clock-icon"
+                src={OrderClockIcon}
+                width="12px"
```
keep in `vw` and `vh` to maintain responsiveness.

sarath-sasi-panickaveetil replied:
Converted to corresponding rem values which is .75rem

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -0,0 +1,299 @@
+          </div>
+          <div className="flex justify-between w-full rounded-b-md border-t py-1 px-3 sm:p-0 border-solid border-[#D4D4D4] bg-[#f4f4f4]">
+            <span className="w-auto sm:w-1/6">{WMS}</span>
+            <span className="w-auto sm:w-5/6 font-medium">{}</span>
```
why empty object ?

sarath-sasi-panickaveetil replied:
removed

sarath-sasi-panickaveetil committed:
PR-11 Review changes

raj-kumar-litmus commented on src/components/orderDetails/OrderDetails.tsx:
```diff
@@ -214,7 +217,7 @@ const OrderDetails: React.FC = () => {
           <div className="flex justify-between w-full rounded-b-md border-t py-1 px-3 sm:p-0 border-solid border-[#D4D4D4] bg-[#f4f4f4]">
             <span className="w-auto sm:w-1/6">{WMS}</span>
-            <span className="w-auto sm:w-5/6 font-medium">{}</span>
+            <span className="w-auto sm:w-5/6 font-medium"></span>
```
why the empty span ?

sarath-sasi-panickaveetil replied:
A value will come here which we don't know at this point. So need to keep a placeholder to accommodate that.

sarath-sasi-panickaveetil committed:
PR-11
> Black theme changes on Cards

sarath-sasi-panickaveetil committed:
Delete src/components/Common directory

Commit:
PR-11 updated

Commit:
update in css

Commit:
Resolved conflicts

# Conflicts:
#	src/components/appcontent.tsx
#	src/components/common/custominputtext.tsx
#	src/components/common/searchbar.tsx
#	src/constants/apiConstants.tsx
#	src/constants/appConstants.tsx
#	src/index.css

raj-kumar-litmus commented on src/@types/BarChart.tsx:
```diff
@@ -21,20 +21,13 @@ export interface ChartDataset {
+  placeholder: string;
+  value: Date;
   onChange: (value: Date) => void;
-  iconPos: "left" | "right";
```
Recheck the lines that u removed here.
I had them for a reason. ex: containerClassName

raj-kumar-litmus commented on src/@types/supportHub.ts:
```diff
@@ -63,15 +63,10 @@ export interface OpmComparisonType {
     y?: any;
   };
   plugins?: any;
-  apiResponse?: any;
```
not sure why these lines are removed.

raj-kumar-litmus commented on src/components/Button.tsx:
```diff
@@ -6,7 +6,6 @@ interface Props {
   isTextButton?: boolean;
   isDisabled?: boolean;
   isRounded?: boolean;
-  id?: string;
```
Lot of my changes are removed.
Please go through all the removed lines and see why they have been removed.
